src/oc/storage/api/boards.clj

```diff
@@ -162,46 +164,54 @@
 
 (defn- create-board [conn {org :existing-org new-board :new-board user :user :as ctx} org-slug]
   (timbre/info "Creating board for org:" org-slug)
-  (if-let [board-result (board-res/create-board! conn new-board)] ; Add the board
-    
-    (let [board-uuid (:uuid board-result)
-          authors (-> ctx :data :authors)
-          viewers (-> ctx :data :viewers)
-          invitation-note (-> ctx :data :note)
-          entries (:entries new-board)
-          notifications (:notifications ctx)]
-      (timbre/info "Created board:" board-uuid "for org:" org-slug)
-      ;; Add any authors specified in the request
-      (doseq [author authors] (add-member conn ctx (:slug org) (:slug board-result) :authors author))
-      ;; Add any viewers specified in the request
-      (doseq [viewer viewers] (add-member conn ctx (:slug org) (:slug board-result) :viewers viewer))
-      ;; Add any entries specified in the request
-      (doseq [entry entries]
-        (let [fixed-entry (-> entry
-                              (assoc :status "published")
-                              (assoc :board-uuid board-uuid))
-              entry-action (if (entry-res/get-entry conn (:uuid entry))
-                             :update
-                             :add)
-              new-entry (if (= entry-action :update)
-                          fixed-entry
-                          (entry-res/->entry conn board-uuid fixed-entry user))]
-          (timbre/info "Upserting entry for new board:" board-uuid)
-          (let [entry-result (entry-res/upsert-entry! conn new-entry user)]
-            (timbre/info "Upserted entry for new board:" board-uuid "as" (:uuid entry-result))
-            (when (= (:status entry-result) "published")
-              (when (= :add entry-action)
-                (entries-api/auto-share-on-publish conn (assoc ctx :existing-board board-result) entry-result))
-              (notification/send-trigger! (notification/->trigger entry-action org board-result {:new entry-result} (:user ctx) nil))))))
-      (let [created-board (if (and (empty? authors) (empty? viewers))
-                            ;; no additional members added, so using the create response is good
-                            board-result
-                            ;; retrieve the board again to get final list of members
-                            (board-res/get-board conn (:uuid board-result)))]
-        (notification/send-trigger! (notification/->trigger :add org {:new created-board :notifications notifications} user invitation-note))
-        {:created-board created-board}))
+  (let [entries (:entries new-board)
+        draft-board? (and (pos? (count entries)) (every? #(-> % :status keyword (= :draft)) entries))
```

Comment:
It seems this code should be moved to the `resources/board/create-board!` function instead of API.

Reply:
Hmm i don't agree, to do so i should pass the list of entries to the `create-board!` function too, this flag is set only when you create a board via this api call in certain conditions (the entries passed in are all drafts).
The `create-board!` function is simply creating a resource in the DB, it has nothing about the logic that validates the data etc.. that's all in the api namespace.
Also if i move the logic that creates the entries there i should move also the add/remove members and all the rest that is in this function.
I'm not saying that your suggestion is not good or doable, but only that it's a bit out of the scope of this.